src/Symfony/Component/VarDumper/Caster/MemcachedCaster.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Caster;
+
+use Symfony\Component\VarDumper\Cloner\Stub;
+
+/**
+ * @author Jan Schädlich <jan.schaedlich@sensiolabs.de>
+ */
+class MemcachedCaster
+{
+    public static function castMemcached(\Memcached $c, array $a, Stub $stub, $isNested)
+    {
+        $a += array(
+            Caster::PREFIX_VIRTUAL.'servers' => $c->getServerList(),
+            Caster::PREFIX_VIRTUAL.'options' => new EnumStub(
+                self::getNonDefaultValueOptions($c)
+            ),
+        );
+
+        return $a;
+    }
+
+    private static function getNonDefaultValueOptions(\Memcached $c)
+    {
+        $options = array(
+            'OPT_COMPRESSION' => self::getOptionValue($c, \Memcached::OPT_COMPRESSION),
+            'OPT_SERIALIZER' => self::getOptionValue($c, \Memcached::OPT_SERIALIZER),
+            'OPT_PREFIX_KEY' => self::getOptionValue($c, \Memcached::OPT_PREFIX_KEY),
+            'OPT_HASH' => self::getOptionValue($c, \Memcached::OPT_HASH),
+            'OPT_DISTRIBUTION' => self::getOptionValue($c, \Memcached::OPT_DISTRIBUTION),
+            'OPT_LIBKETAMA_COMPATIBLE' => self::getOptionValue($c, \Memcached::OPT_LIBKETAMA_COMPATIBLE),
+            'OPT_BUFFER_WRITES' => self::getOptionValue($c, \Memcached::OPT_BUFFER_WRITES),
+            'OPT_BINARY_PROTOCOL' => self::getOptionValue($c, \Memcached::OPT_BINARY_PROTOCOL),
+            'OPT_NO_BLOCK' => self::getOptionValue($c, \Memcached::OPT_NO_BLOCK),
+            'OPT_TCP_NODELAY' => self::getOptionValue($c, \Memcached::OPT_TCP_NODELAY),
+            'OPT_SOCKET_SEND_SIZE' => self::getOptionValue($c, \Memcached::OPT_SOCKET_SEND_SIZE),
+            'OPT_SOCKET_RECV_SIZE' => self::getOptionValue($c, \Memcached::OPT_SOCKET_RECV_SIZE),
+            'OPT_CONNECT_TIMEOUT' => self::getOptionValue($c, \Memcached::OPT_CONNECT_TIMEOUT),
+            'OPT_RETRY_TIMEOUT' => self::getOptionValue($c, \Memcached::OPT_RETRY_TIMEOUT),
+            'OPT_SEND_TIMEOUT' => self::getOptionValue($c, \Memcached::OPT_SEND_TIMEOUT),
+            'OPT_RECV_TIMEOUT' => self::getOptionValue($c, \Memcached::OPT_RECV_TIMEOUT),
+            'OPT_POLL_TIMEOUT' => self::getOptionValue($c, \Memcached::OPT_POLL_TIMEOUT),
+            'OPT_CACHE_LOOKUPS' => self::getOptionValue($c, \Memcached::OPT_CACHE_LOOKUPS),
+            'OPT_SERVER_FAILURE_LIMIT' => self::getOptionValue($c, \Memcached::OPT_SERVER_FAILURE_LIMIT),
+        );
+
+        return array_filter($options, function ($option) {
+            return null !== $option ?? $option;
+        });
+    }
+
+    private static function getOptionValue(\Memcached $c, int $option)
```

Comment:
I'd like to suggest a completely different implementation: instead of hardcoding the list of constants we could use reflection to discover them and be compatible with any new one that could be added in the future.
And the default values, we could also discover them by creating an instance of Memcached (since the constructor doesn't initiate the connection)
The result of this could be an array stored once in a static property.
WDYT?

Reply:
Please have a look, if I got you right.